Update native-aot.md (#29015)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/native-aot.md

</PropertyGroup>
```

> A project that uses native AOT publishing uses JIT compilation when running locally, but there are some observable differences:

Author: ```suggestion
A project that uses native AOT publishing uses JIT compilation when running locally. The AOT app and JIT app have the following differences:
```

> * Some features that aren't compatible with native AOT are disabled and throw exceptions at runtime.

Author: ```suggestion
* Features that aren't compatible with native AOT are disabled and throw exceptions at runtime.
```

</PropertyGroup>
```

> A project that uses native AOT publishing uses JIT compilation when running locally. The AOT app and JIT app have the following differences:

Author: ```suggestion
A project that uses native AOT publishing uses JIT compilation when running locally. The AOT app has the following differences:
```

> * Features that aren't compatible with native AOT are disabled and throw exceptions at runtime.

Reviewer: I think the original statement of "Some features" is probably more accurate. We've tried to make it so that things don't silently fail, but I don't think we are at 100% yet. If we are writing this as it should be for .NET 8.0 RTM then yes, this is fine.

</Project>
```

Run the `dotnet build` command. `publish` isn't necessary to view generated code. The built output contains an `obj/Debug/net8.0/generated/` directory and all the generated files for the project.

> When `dotnet publish` is run:
> 
> * The project source files and generated source files are compiled.
> * Generated assemblies are passed to an native IL compiler which produces the native executable. The native executable contains the native machine code.
> * The native app is ready to run.

Author: ```suggestion
The `dotnet publish` command:

* Compiles the source files.
* Generates files which are are compiled.
* Passes generated assemblies to an native IL compiler. The IL compiler produces the native executable. The native executable contains the native machine code.
```

```suggestion
When `dotnet publish` is run:

* The project source files and generated source files are compiled.
* Generated assemblies are passed to an native IL compiler which produces the native executable. The native executable contains the native machine code.
* The native app is ready to run.
```

Author: I prefer this, please accept this suggestion.

Reviewer: I would conflate the first two bullet points since the source generated files and the user-provided source files are compiled at the same time. There isn't two different compiation steps to go from C# code to DLL. There is a second IL compilation step to go from DLL/IL to native executable.

  * Improved deployment where container orchestrators need manage transition from one version of the app to another.
* **Reduce memory demand**: Native AOT apps can have reduced memory demands depending on the work being performed by the app. Reduced memory consumption can lead to greater deployment density and improved scalability.

The template app was run in our benchmarking lab and shows the following improvements in size, memory, and startup time:

![Chart showing comparison of application size, memory use and startup time metrics of an AOT published app, a runtime app that is trimmed, and an untrimmed runtime app.](~/fundamentals/aot/_static/aot-runtime-trimmed-perf-chart.png)

> The preceding chart shows the native AOT has a significantly lower app size, memory used, and startup time.

Reviewer: memory used => memory usage?

> The preceding chart shows the native AOT has a significantly lower app size, memory used, and startup time.

Author: ```suggestion
The preceding chart shows the native AOT has a significantly lower app size, memory usage, and startup time.
```
